src/backend/parser/cypher_gram.y

```diff
@@ -594,6 +599,53 @@ updating_clause:
     | merge
     ;
 
+subquery_stmt:
+    single_subquery
+        {
+            $$ = $1;
+        }
+    | subquery_stmt UNION all_or_distinct subquery_stmt
+        {
+            $$ = list_make1(make_set_op(SETOP_UNION, $3, $1, $4));
+        }
+    ;
+
+subquery_stmt_no_return:
+    single_subquery_no_return
+        {
+            $$ = $1;
+        }
+    | subquery_stmt_no_return UNION all_or_distinct subquery_stmt_no_return
+        {
+            $$ = list_make1(make_set_op(SETOP_UNION, $3, $1, $4));
+        }
+    ;
+
+single_subquery:
+        subquery_part_init reading_clause_list return
+        {
+            $$ = list_concat($1, lappend($2, $3));  
+        }
+    ;
+
+single_subquery_no_return:
+        subquery_part_init reading_clause_list
```

Comment:
extra indentation?

Reply:
Thanks for pointing that out~ I amended the commit. Feel free to merge it whenever you see fit